### PR TITLE
docs:add a ingress blog

### PR DESCRIPTION
**Pull request by KID-G**
Changes:

Add a Apache APISIX Ingress Controller about How to support Custom Plugins blog。

### CODE REVIEW

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+**在实际的生产环境中，客户端访问的需求是多种多样的。比如最常见的认证、路由重写等能力，通过 Ingress 资源是无法直接进行描述的。那么这些需求要如何满足呢？**
+
+## Ingress-NGINX 如何支持扩展功能
+
+首先我以 Kubernetes 社区的 Ingress-NGINX controller 为例，介绍如何在其中使用扩展功能。
+
+在 Ingress-NGINX 项目中，可以为 Ingress 资源增加一些 Annotation 来描述其需要使用的扩展能力。比如使用如下配置便可开启 cors 能力。
+
+```plain
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]:
```suggestion
```yaml
```

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+![output.png](https://static.apiseven.com/2022/12/14/63996c4d6f23e.png)
+
+目前，在 Apache APISIX 中支持多种方式进行自定义插件的开发：
+
+* 使用 Lua 进行插件的开发，这类插件会在 APISIX 内部运行；
+* 使用其他语言进行插件的开发，这种机制叫作 Plugin Runner，利用该机制开发的插件属于 external-plugin。
+
+关于 APISIX 插件的开发，可参考官方文档：
+
+* 插件开发 [https://apisix.apache.org/docs/apisix/plugin-develop/](https://apisix.apache.org/docs/apisix/plugin-develop/)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]:
```suggestion
* [插件开发](https://apisix.apache.org/docs/apisix/plugin-develop/)
```

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+* External Plugin 开发：[https://apisix.apache.org/docs/apisix/external-plugin/](https://apisix.apache.org/docs/apisix/external-plugin/)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]:
```suggestion
* [External Plugin 开发](https://apisix.apache.org/docs/apisix/external-plugin/)
```

**Review by SylviaBABY — DISMISSED** [review]

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+<!--truncate-->
+
+> 作者：张晋涛，API7.ai 云原生技术专家，Apache APISIX PMC 成员，Apache APISIX Ingress Controller 项目维护者。
+
+## Ingress 和 Ingress controller
+
+Kubernetes 中的 Ingress 是一种资源对象，用于定义如何从 Kubernetes 集群外访问到 Kubernetes 集群内的服务，其中包含了具体的访问规则，通常情况下客户端使用 HTTP/HTTPS 协议进行访问。
+
+客户端可按照 Ingress 资源定义的规则，将客户端请求路由到 Kubernetes 集群中的服务或具体的 Pod中。
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]:
```suggestion
客户端可按照 Ingress 资源定义的规则，将客户端请求路由到 Kubernetes 集群中的服务或具体的 Pod 中。
```

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+tags: [Ecosystem]
+---
+
+> 本篇主要介绍了 Ingress 资源相关的语义，以及如何对 Ingress 资源进行能力的扩展。
+
+<!--truncate-->
+
+> 作者：张晋涛，API7.ai 云原生技术专家，Apache APISIX PMC 成员，Apache APISIX Ingress Controller 项目维护者。
+
+## Ingress 和 Ingress controller
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]: https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/

File: blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+      - path: /apisix-ingress
+        pathType: Exact
+        backend:
+          service:
+            name: apisix-gateway
+            port:
+              number: 80
+```
+
+上述配置在 Ingress 资源中增加了 cors 相关的一些信息。APISIX Ingress controller 可以识别这些信息，并将这些信息转换为数据面中 cors 的配置，进而完成对 Ingress 资源的扩展。
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/15/how-support-ingress-custom-plugins.md]:
```suggestion
上述配置在 Ingress 资源中增加了 `cors` 相关的一些信息。APISIX Ingress controller 可以识别这些信息，并将这些信息转换为数据面中 `cors` 的配置，进而完成对 Ingress 资源的扩展。
```

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Pull request merged**